micro-rdk/src/common/data_manager.rs

```diff
@@ -327,8 +398,10 @@ where
                     if next_message.is_empty()
                         || ((next_message.len() + current_chunk_size) > MAX_SENSOR_CONTENTS_SIZE)
                     {
-                        let data: Result<Vec<SensorData>, prost::DecodeError> =
-                            current_chunk.drain(..).map(SensorData::decode).collect();
+                        let data: Result<Vec<SensorData>, DataSyncError> = current_chunk
+                            .drain(..)
+                            .map(time_corrected_reading_handler(self.robot_start_time))
```

Comment:
What happens here if the function returned by time_corrected_reading_handler returns an error?